addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4278,26 +4285,31 @@ registry.ImageTools = SnippetOptionWidget.extend({
         new weWidgets.ImageCropWidget(this, this.$target[0]).appendTo(this.options.wysiwyg.$editable);
     },
     transform() {
-        if (this.$target.data('transfo-destroy')) {
-            this.$target.removeData('transfo-destroy');
-            return;
-        }
-        const document = this.$target[0].ownerDocument
+        const document = this.$target[0].ownerDocument;
         this.$target.transfo({ document });
+        $('[data-transform]').toggleClass('active', this.$target.is('[style*="transform"]'));
+        const mouseup = () => {
+            this._toggleRotateButton();
+        };
         const mousedown = mousedownEvent => {
             if (!$(mousedownEvent.target).closest('.transfo-container').length) {
                 this.$target.transfo('destroy');
                 $(document).off('mousedown', mousedown);
+                $(document).off('mouseup', mouseup);
             }
-            if ($(mousedownEvent.target).closest('#image-transform').length) {
-                this.$target
-                    .data('transfo-destroy', true)
-                    .attr('style', (this.$target.attr('style') || '')
-                    .replace(/[^;]*transform[\w:]*;?/g, ''));
+            if ($(mousedownEvent.target).closest('[data-transform]').length) {
+                this.reset();
             }
             this.$target.trigger('content_changed');
         };
         $(document).on('mousedown', mousedown);
+        $(document).on('mouseup', mouseup);
```

Comment:
All of this seems wrong, I am not sure to even understand it. Option methods are called with 3 parameters previewMode, widgetValue, params. If the spec ask us to handle double click on a we-button... this must be a new feature of the button widget. Currently we call methods defined on the button once we click on it. We may define a syntax for methods to call when doubleclick on it.

I wonder if this is what we want though... the spec is not clear. It says that double clicking is not intuitive but still wants to keep it :shrug: I think you should re-discuss it with SBU.